[clang-tidy] Moved Multiple Inheritence check from fuchsia to misc module

Bhuvan1527 opened the pull request:
Resolves: [#171136](https://github.com/llvm/llvm-project/issues/171136) 

Referred the issue tagged in the issue mentioned.

Bhuvan1527 committed:
[clang-tidy] Moved Multiple Inheritence check from fuchsia to misc module

Resolves: [#171136](https://github.com/llvm/llvm-project/issues/171136)
Referred the issue tagged in the issue mentioned.

EugeneZelenko commented on clang-tools-extra/clang-tidy/misc/MiscTidyModule.cpp:
```diff
@@ -86,6 +87,8 @@ class MiscModule : public ClangTidyModule {
         "misc-use-anonymous-namespace");
     CheckFactories.registerCheck<UseInternalLinkageCheck>(
         "misc-use-internal-linkage");
+    CheckFactories.registerCheck<MultipleInheritanceCheck>(
```
Please keep alphabetical order.

EugeneZelenko reviewed (commented):
Documentation was not moved and Release Notes entry is missing.

zeyi2 reviewed (changes requested):
The code change looks nice. However you still need to update the documentations.

more specifically:
- Edit `fuchsia/multiple-inheritance.rst`
- Create a new doc for `misc/multiple-inheritance.rst`
- Modify `list.rst` for correct aliasing
- Document the change in `clang-tools-extra/docs/ReleaseNotes.rst`

You can use [this PR](https://github.com/llvm/llvm-project/pull/171058) as a style reference.

Bhuvan1527 committed:
[clang-tidy] Moved Multiple Inheritence check from fuchsia to misc module

Updated the documentation in the commit,
1. ReleaseNotes.rst
2. respective docs in the docs/clang-tidy/checks/
3. list.rst

Bhuvan1527 commented on clang-tools-extra/clang-tidy/misc/MultipleInheritanceCheck.h:
```diff
@@ -6,12 +6,12 @@
 #include "../ClangTidyCheck.h"
 
-namespace clang::tidy::fuchsia {
+namespace clang::tidy::misc {
```
Addressed in the latest commit

vbvictor commented on clang-tools-extra/clang-tidy/fuchsia/CMakeLists.txt:
Please include misc Module here - otherwise will lead to linking errors

zeyi2 reviewed (commented):
May be a good idea to add reference here.

Bhuvan1527 commented on clang-tools-extra/docs/clang-tidy/checks/fuchsia/multiple-inheritance.rst:
```diff
@@ -3,44 +3,7 @@
-  };
-
-See the features disallowed in Fuchsia at https://fuchsia.dev/fuchsia-src/development/languages/c-cpp/cxx?hl=en
+See the features disallowed in Fuchsia at https://fuchsia.dev/fuchsia-src/development/languages/c-cpp/cxx?hl=en
```
Addressed in the latest commit

Bhuvan1527 commented on clang-tools-extra/docs/clang-tidy/checks/misc/multiple-inheritance.rst:
```diff
@@ -0,0 +1,46 @@
+    virtual int foo() override { return 0; }
+    virtual int bar() override { return 0; }
+  };
+
```
Addressed in the latest commit

vbvictor reviewed (approved):
lgtm with nit

zeyi2 commented on clang-tools-extra/docs/clang-tidy/checks/fuchsia/multiple-inheritance.rst:
```diff
@@ -3,44 +3,5 @@
-
-See the features disallowed in Fuchsia at https://fuchsia.dev/fuchsia-src/development/languages/c-cpp/cxx?hl=en
+The `fuchsia-multiple-inheritance` check is an alias, please See
+:doc:`misc-multiple-inheritance <../misc/multiple-inheritance>` for details.
```
```suggestion
:doc:`misc-multiple-inheritance <../misc/multiple-inheritance>` for details.

See the features disallowed in Fuchsia at https://fuchsia.dev/fuchsia-src/development/languages/c-cpp/cxx?hl=en
```

zeyi2 reviewed (approved):
LGTM, thanks!

Also, you can apply vbvictor's suggestion like this.

EugeneZelenko commented on clang-tools-extra/docs/clang-tidy/checks/fuchsia/multiple-inheritance.rst:
```diff
@@ -3,44 +3,5 @@
-    virtual int bar() override { return 0; }
-  };
-
-See the features disallowed in Fuchsia at https://fuchsia.dev/fuchsia-src/development/languages/c-cpp/cxx?hl=en
```
Somehow, lines are still in document.

Bhuvan1527 replied:
Sorry @EugeneZelenko , I didn't understand your comment. 

I believe @vbvictor suggested not to remove the html link from this file. So I preserved it.
